pkg/minikube/update/update.go

```diff
@@ -175,41 +182,77 @@ func updateBinary(v semver.Version, downloadBinary, updateLinkPrefix, downloadLi
 }
 
 func updateBinaryFile(url string, checksum []byte) {
-	fmt.Println("Downloading updated binary")
+	fmt.Println("Downloading updated tarball")
 	httpResp, err := http.Get(url)
 	if err != nil {
-		glog.Errorf("Cannot download binary: %s", err)
+		glog.Errorf("Cannot download tarball: %s", err)
 		os.Exit(1)
 	}
 	defer func() { _ = httpResp.Body.Close() }()
 
-	binary := httpResp.Body
+	tarball := httpResp.Body
 	if httpResp.ContentLength > 0 {
 		bar := pb.New64(httpResp.ContentLength).SetUnits(pb.U_BYTES)
 		bar.Start()
-		binary = bar.NewProxyReader(binary)
+		tarball = bar.NewProxyReader(tarball)
 		defer func() {
 			<-time.After(bar.RefreshRate)
 			fmt.Println()
 		}()
 	}
-	err = update.Apply(binary, update.Options{
-		Hash:     crypto.SHA256,
-		Checksum: checksum,
+
+	// verify checksum
+	tarballBytes, err := ioutil.ReadAll(tarball)
+
+	tarballChecksum, err := checksumFor(crypto.SHA256, tarballBytes)
+
+	if !bytes.Equal(tarballChecksum, checksum) {
+		fmt.Println(fmt.Errorf("Updated file has wrong checksum. Expected: %x, got: %x", tarballChecksum, checksum))
+		os.Exit(1)
+	}
+
+	// Write tar file to disk
+	err = ioutil.WriteFile("/tmp/minishift.tgz", tarballBytes, 0644)
```

Comment:
In fact, I think we should use the tmp directory of the MINISHIFT_HOME to do all this, e.g. _ ~/.minishift/tmp/_